script.js

```diff
@@ -1,4 +1,309 @@
+//Add global states to include wins and draws
+var playerScore = 0;
+var comScore = 0;
+var drawScore = 0;
+
+//Add states for username and game mode
+var userName = "";
+var gameMode = "";
+
+//Game mode checker
+var gameModeState = 0;
+
+//Add WinState for korean mode
+var playerLastWinState = 0;
+var comLastWinState = 0;
+var winnerState = 0;
+
 var main = function (input) {
-  var myOutputValue = 'hello world';
+  //Convert input to lowercase and remove white space, allow for minor typos
+  var convertInput = input.toLowerCase().replace(/\s/g, "");
+
+  //Prints error message if input is not SPS or reversed SPS
+  var myOutputValue = `Please enter 'stone', 'paper' or 'scissors' <br>Current Score<br>User:${playerScore}<br>Com:${comScore}<br>Draws:${drawScore}`;
+
+  //If userName is not set, use first input as username
+  if (userName == "") {
+    if (input == "") {
+      return `Please input something for your username`;
+    }
+
+    userName = input;
+
+    return `Hi ${userName}, Please enter game mode to start playing.`;
+  }
+
+  //If mode is not selected, prompt to choose mode
+  if (gameModeState == 0 && gameMode == "") {
+    if (
+      input != "normal" &&
+      input != "reversed" &&
+      input != "korean" &&
+      input != "computer" &&
+      input != "word" &&
+      input != "dice"
+    ) {
+      return `Please select 'dice', 'word', 'normal' ,'reversed','computer' or 'korean' mode`;
```

Comment:
There is something know as [SCREAM_CASE constants ](https://www.freecodecamp.org/news/when-to-capitalize-your-javascript-constants-4fabc0a4a4c4/)

I can touch more on this as the days go by, but it's simply another way to name variables that remain unchanged through out the program eg.
```
var NORMAL_MODE = "normal"
var REVERSED_MODE = "reversed"
var KOREAN_MODE = "korean"
```
So these can be declared as global variables at the top as well